bin/test.c

```diff
@@ -130,6 +130,9 @@ void test_parser() {
       0x30, 0x17, 0x00, 0x0b, 0x53, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x54, 0x6f, 0x70, 0x69, 0x63, 0x48, 0x65, 0x6c, 0x6c,
       0x6f, 0x20, 0x4d, 0x51, 0x54, 0x54,
 
+      /* Publish without payload */
+      0x30, 0x0d, 0x00, 0x0b, 'S', 'a', 'm', 'p', 'l', 'e', 'T', 'o', 'p', 'i', 'c',
```

Comment:
One last thing -  I understand that this is just a sample of MQTT message without payload but it is a valid one? Is the MQTT protocol allowing publishing to a topic an empty message? If not - please, replace the example with a valid MQTT message that does not have payload.

Reply:
Yes, this one is valid. See http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718040
> The length of the payload can be calculated by subtracting the length of the variable header from the Remaining Length field that is in the Fixed Header. It is valid for a PUBLISH Packet to contain a zero length payload.

Reply:
Perfect, thanks.